test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,139 @@
+		o.Expect(err).ToNot(o.HaveOccurred())
+		initialEtcdPodCount = len(etcdPods)
+
+		masterNodes := masterNodes(oc)
+		etcdTargetNodeName = masterNodes[0].Name
+	})
+
+	g.AfterEach(func(ctx context.Context) {
+
+		// Reset the log level back to initial log level to leave the cluster in the same state.
+		g.GinkgoT().Log("resetting the log level back to intial log level")
+		data := fmt.Sprintf(`{"spec": {"logLevel": "%s"}}`, initialLogLevel)
+		_, err := oc.AdminOperatorClient().OperatorV1().Etcds().Patch(ctx, "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("debugging into the same node to add back the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mv /var/lib/etcd-backup/etcd-pod.yaml /etc/kubernetes/manifests/").Execute()
+		o.Expect(err).ToNot(o.HaveOccurred(), "failed to add etcd static pod back to the node")
+
+		g.GinkgoT().Log("waiting for all the etcd instances to be available to return the cluster to its original state")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount)
+
+	})
+
+	g.It("is able to block the rollout of a revision when the quorum is not safe", func(ctx context.Context) {
+		var err error
+
+		g.GinkgoT().Log("debugging into the node to remove the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mkdir -p /var/lib/etcd-backup && mv /etc/kubernetes/manifests/etcd-pod.yaml /var/lib/etcd-backup").Execute()
+		err = errors.Wrap(err, "failed to remove etcd static pod from the node")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("waiting for the etcd pod to be removed")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount-1)
+
+		g.GinkgoT().Log("ensuring the quorum guard dependent controllers have degraded")
+		conditionTypes := []string{
+			"EtcdCertSignerControllerDegraded",
+			"EtcdEndpointsDegraded",
+			"TargetConfigControllerDegraded",
+		}
+		o.Expect(wait.PollUntilContextTimeout(ctx, 10*time.Second, 5*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+			// retrieve the operator status
+			etcdCluster, err := oc.AdminOperatorClient().OperatorV1().Etcds().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil {
+				klog.Infof("error while getting operator status: %v", err)
+				return false, nil
+			}
+
+			for _, conditionType := range conditionTypes {
+				isConditionTrue := librarygov1helpers.IsOperatorConditionTrue(etcdCluster.Status.Conditions, conditionType)
+				if !isConditionTrue {
+					return false, nil
+				}
+			}
+			return true, nil
+		})).ToNot(o.HaveOccurred(), "expected the quorum guard dependent controllers to be degraded")
```

Comment:
Can we list the condition status for all 3 controllers here to show which one isn't yet degraded. If this test ever fails on this assertion then that will make it more obvious as to what controller(s) didn't go degraded before the timeout.

Reply:
that's a good one, I had to dig into the CEO logs last time to get this

Reply:
Have added the code to list the controllers that hasn't yet degraded. The code will now display the error in the following format:
```
fail [github.com/openshift/origin/test/extended/dr/static_pod_rollouts.go:110]: expected the quorum guard dependent controllers to be degraded; however, the following indicated by their condition types have not: [TargetConfigControllerDegraded]
Unexpected error:
    <context.deadlineExceededError>: 
    context deadline exceeded
    
        {}
occurred
Ginkgo exit error 1: exit with code 1
```
Will this be sufficient ?